llvm-bazel/overlay_directories.py

```diff
@@ -0,0 +1,83 @@
+#!/bin/python3
+
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Overlays two directories into a target directory using symlinks.
+
+Tries to minimize the number of symlinks created (that is, does not symlink
+every single file). Symlinks every file in the overlay directory. Only symlinks
+individual files in the source directory if their parent directory is also
+contained in the overlay directory tree.
+"""
+
```

Comment:
Might be worth adding:

```
from __future__ import print_function
```

And then including a little inline version check for python > 3:

```
import sys
if sys.version_info[0] < 3:
  raise RuntimeError("Must be invoked with a python 3 interpreter but was %s" % sys.executable)
```

That may be enough to parse and do an inline version check on python 2.7 without triggering other failures first. While we probably don't care much about python 2.7, it is still the default on a lot of supported OS's and if possible to be verbose about the mismatch, it can save a lot of time.

Reply:
I don't appear to even need the print_function import (even when the script contains a print usage). Invoking with python2 gives the expected error message like this.

Reply:
Oh right - without that Python2 will parse a print function as either a print statement with a parenthesized operand or a print statement of a tuple (if multiple operands). It will break with any named arguments, iirc. But since you just want the file to parse without error so it can execute the first couple of lines, it is fine either way.